virtool/analyses/data.py

```diff
@@ -47,79 +46,160 @@
+                            {
+                                "$lookup": {
+                                    "from": "jobs",
+                                    "localField": "job.id",
+                                    "foreignField": "_id",
+                                    "as": "job",
+                                }
+                            },
+                            {
+                                "$unwind": {
+                                    "path": "$job",
+                                    "preserveNullAndEmptyArrays": True,
+                                }
+                            },
+                            {
+                                "$lookup": {
+                                    "from": "subtraction",
+                                    "localField": "subtractions",
+                                    "foreignField": "_id",
+                                    "as": "subtractions",
+                                }
+                            },
+                            {
+                                "$lookup": {
+                                    "from": "references",
+                                    "localField": "reference.id",
+                                    "foreignField": "_id",
+                                    "as": "reference",
+                                }
+                            },
+                            {"$unwind": {"path": "$reference"}},
+                            {
+                                "$lookup": {
+                                    "from": "users",
+                                    "localField": "user.id",
+                                    "foreignField": "_id",
+                                    "as": "user",
+                                }
+                            },
+                            {"$unwind": {"path": "$user"}},
+                            {"$sort": sort},
+                            {"$skip": skip_count},
+                            {"$limit": per_page},
+                        ],
+                    }
+                },
+                {
+                    "$project": {
+                        "data": dict_projection,
+                        "total_count": {
+                            "$arrayElemAt": ["$total_count.total_count", 0]
+                        },
+                        "found_count": {
+                            "$arrayElemAt": ["$found_count.found_count", 0]
+                        },
+                    }
+                },
+            ],
+        ):
+            data = paginate_dict["data"]
+            found_count = paginate_dict.get("found_count", 0)
+            total_count = paginate_dict.get("total_count", 0)
+
+        for analysis in data:
+            analysis["job"] = (
+                await processor(self._db, analysis["job"])
+                if "job" in analysis
+                else None
+            )
```

Comment:
This is going to kill performance after you have spent all this time optimizing it.

Can this not also be done in the aggregation?